clang/include/clang/Basic/Builtins.td

```diff
@@ -4883,6 +4882,12 @@ def HLSLRadians : LangBuiltin<"HLSL_LANG"> {
   let Prototype = "void(...)";
 }
 
+def HLSLBufferUpdateCounter : LangBuiltin<"HLSL_LANG"> {
+  let Spellings = ["__builtin_hlsl_buffer_update_counter"];
+  let Attributes = [NoThrow];
+  let Prototype = "uint32_t(...)";
```

Comment:
Why does this look like it's overloaded?  Shouldn't this have a specific signature like `"uint32_t(__hlsl_resource_t, int)"`?

Reply:
It is overloaded. The intrinsic takes `__hlsl_resource_t` decorated with a resource type attributes and every unique combination of the attributes means a different resource type. And I don't think the attributes can be expressed here anyway even if there was just one overload.